apis/pipelines/v1alpha6/runconfiguration_conversion.go

```diff
@@ -50,9 +54,58 @@ func (dst *RunConfiguration) ConvertFrom(srcRaw conversion.Hub) error {
 	dst.Status.Provider = src.Status.Provider.Name
 	remainder.ProviderNamespace = src.Spec.Run.Provider.Namespace
 	remainder.ProviderStatusNamespace = src.Status.Provider.Namespace
+	dst.Spec.Run.RuntimeParameters = convertRuntimeParametersFrom(src.Spec.Run.Parameters)
 
 	dst.TypeMeta.APIVersion = dstApiVersion
 	dst.Status.SynchronizationState = src.Status.Conditions.GetSyncStateFromReason()
 
 	return pipelines.SetConversionAnnotations(dst, &remainder)
 }
+
+func (v *ValueFrom) convertToHub() *hub.ValueFrom {
+	if v != nil {
+		return &hub.ValueFrom{
+			RunConfigurationRef: hub.RunConfigurationRef{
+				Name:           v.RunConfigurationRef.Name,
+				OutputArtifact: v.RunConfigurationRef.OutputArtifact,
+			},
+		}
+	}
+	return nil
+}
+
+func convertFromHubValueFrom(v *hub.ValueFrom) *ValueFrom {
+	if v != nil {
+		return &ValueFrom{
+			RunConfigurationRef: RunConfigurationRef{
+				Name:           v.RunConfigurationRef.Name,
+				OutputArtifact: v.RunConfigurationRef.OutputArtifact,
+			},
+		}
+	}
+	return nil
+}
+
+func convertRuntimeParametersTo(rtp []RuntimeParameter) []hub.Parameter {
+	var hubParams []hub.Parameter
+	for _, namedValue := range rtp {
+		hubParams = append(hubParams, hub.Parameter{
+			Name:      namedValue.Name,
+			Value:     namedValue.Value,
+			ValueFrom: namedValue.ValueFrom.convertToHub(),
+		})
+	}
+	return hubParams
+}
+
+func convertRuntimeParametersFrom(hubParams []hub.Parameter) []RuntimeParameter {
+	var rtp []RuntimeParameter
+	for _, namedValue := range hubParams {
+		rtp = append(rtp, RuntimeParameter{
+			Name:      namedValue.Name,
+			Value:     namedValue.Value,
+			ValueFrom: convertFromHubValueFrom(namedValue.ValueFrom),
+		})
+	}
+	return rtp
```

Comment:
```suggestion
	rtp := []RuntimeParameter{}
	for _, namedValue := range hubParams {
		rtp = append(rtp, RuntimeParameter{
			Name:      namedValue.Name,
			Value:     namedValue.Value,
			ValueFrom: convertFromHubValueFrom(namedValue.ValueFrom),
		})
	}
	return rtp
```

Reply:
Not sure if we need to defend against Nil slice in this instance, this would prevent nil being returned by instantiating the slice from the beginning.

Reply:
Not 100% sure either, but it seems safer to return an empty array instead, so I've made the change in 4e36414.